### ISSUE TITLE
Support introspection of ParseError?

### ISSUE THREAD

**Issue by d-e-s-o** [chronotope/chrono#319]
I think it would be great to have some way to poke into `ParseError` to see what kind of error it actually is. Right now the only possibility I see is to compare strings, which is obviously not something I am willing to do. The easiest way to accomplish that would be to make `ParseErrorKind` part of the public interface.

One use case:

A user wants to parse dates, but there are different formats that the incoming data could have. I'd want to let `chrono` take care of the parsing and then tell me that parsing failed, and not something else. And only if that is the case move on to trying a different format. Basically:

```rust
match DateTime::parse_from_str(..., fmts[i]) {
  Ok(dt) => return dt,
  Err(err) if err.kind() == ParseKindError::Invalid {
    continue
  },
  Err(err) => panic("{}", err),
}
```

That's just one example. I am sure there are more. In fact, error types in most crates can be inspected like this and matched on, at least in some form. Also see `std::io::Error` and its `ErrorKind` (which I guess `ParseError` may have been inspired by).

**Comment by quodlibetor** [comment]
Yes, ParseError and the string representations of the errors are terrible, I would love to improve this, unfortunately it's both a fairly invasive change to the parsing code and a backwards-incompatible one. It is highest on my list of priorities other than breaking chrono up into multiple crates, though.
